Parsers/ASimAuthentication/Parsers/ASimAuthenticationCiscoIse.yaml

```diff
@@ -0,0 +1,84 @@
+      | extend 
+          EventVendor = "Cisco"
+          , EventProduct = "ISE"
+          , EventProductVersion = "3.2"
+          , EventCount = int(1)
+          , EventSchema = "Authentication"
+          , EventSchemaVersion = "0.1.3"
+          , EventStartTime = coalesce(EventTime, TimeGenerated)
+          , EventEndTime = coalesce(EventTime, TimeGenerated)
+          , EventOriginalSeverity = extract(@".*(NOTICE|INFO|WARN|WARNING|ERROR|FATAL|DEBUG).*", 1, SyslogMessage)
+      | lookup EventName on EventOriginalType 
+      | parse-kv SyslogMessage as (FailureReason:string, NetworkDeviceName:string, Protocol:string, DestinationIPAddress:string, DestinationPort:int, ['User-Name']:string, UserName:string, User:string, SessionID:string, ['Remote-Address']:string, ['Device IP Address']:string, ['Device Port']:int, ["cisco-av-pair=audit-session-id"]:string) with (pair_delimiter=',', kv_delimiter='=')
+      | project-rename
+          LogonProtocol=Protocol
+          , TargetIpAddr=DestinationIPAddress
+          , TargetPortNumber=DestinationPort
+          , TargetSessionId=["cisco-av-pair=audit-session-id"]
+          , SrcPortNumber=['Device Port']
+      | extend DvcHostname = coalesce(NetworkDeviceName, Computer, HostName)
+      | extend TargetUsername = coalesce(['User-Name'], UserName, User)
+      | extend TargetUsernameType = ASIM_GetUsernameType(TargetUsername)
+          , SrcIpAddr = coalesce(['Device IP Address'], ['Remote-Address'], tostring(extract(@"Caller-Station-ID=(\d{1,3}\.\d{1.3}\.\d{1,3}\.\d{1,3})", 1, SyslogMessage)), "")
+      | extend EventOriginalResultDetails = case(isnotempty(FailureReason), FailureReason, EventOriginalResultDetails)
+      | project-away FailureReason, NetworkDeviceName, ['User-Name'], UserName, User, ['Remote-Address'], ['Device IP Address']
+      | extend DvcIpAddr = extract(@"(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})", 1, Computer)
+      | extend SrcHostname = ASIM_GetSrcHostname(SyslogMessage)
```

Comment:
Use In-built version of ASIM_GetSrcHostname

Reply:
fixed